scylla/src/lib.rs

```diff
@@ -230,6 +230,13 @@ pub mod deserialize {
             UdtIterator, UdtTypeCheckErrorKind,
         };
     }
+
+    // Shorthands for better readability.
+    #[cfg_attr(not(test), allow(unused))]
+    pub(crate) trait DeserializeOwnedValue: for<'r> DeserializeValue<'r, 'r> {}
+    impl<T> DeserializeOwnedValue for T where T: for<'r> DeserializeValue<'r, 'r> {}
+    pub(crate) trait DeserializeOwnedRow: for<'r> DeserializeRow<'r, 'r> {}
+    impl<T> DeserializeOwnedRow for T where T: for<'r> DeserializeRow<'r, 'r> {}
```

Comment:
Question regarding this bounds (applicable also before this commit): Why use a single lifetime?
In other words, why `for<'r> DeserializeValue<'r, 'r>` and not `for<'frame, 'metadata> DeserializeValue<'frame, 'metadata>`? I'm not saying it is wrong - I just wanted to understand the reasoning.

The second bound is more restrictive, right? Maybe the first one is just enough, or maybe there is no difference?

Reply:
The first one is just enough.

Meanwhile, the second one is too restrictive, and I believe impossible to be satisfied!
Namely,
```rust
for<'frame, 'metadata> DeserializeValue<'frame, 'metadata>
```
would require that `DeserializeValue` is implemented for all possible ('frame, 'metadata) pairs, including those where `'frame: 'metadata` does not hold true - and this contradicts the said logical lifetime relationship that normal `DeserializeValue` implementors impose.

Reply:
As proved experimentally, `for<'frame, 'metadata> DeserializeValue<'frame, 'metadata>` works, and those traits could use that.

Given that they are internal and not exposed to user, it does not matter what they use.